controllers/news.controller.js

```diff
@@ -1,36 +1,56 @@
-const { selectTopics, selectArticleById, selectArticles } = require("../models/news.model");
-const endpointData =  require(`${__dirname}/../endpoints.json`)
+const {
+  selectTopics,
+  selectArticleById,
+  selectArticles,
+  checkArticleExists,
+  selectCommentsById,
+} = require("../models/news.model");
+const endpointData = require(`${__dirname}/../endpoints.json`);
 
 exports.getTopics = (req, res, next) => {
-  selectTopics()
-    .then((topics) => {
-      res.status(200).send({ topics });
-    })
+  selectTopics().then((topics) => {
+    res.status(200).send({ topics });
+  });
 };
 
 exports.getEndpoints = (req, res, next) => {
-  return res.status(200).send({endpointData});
+  res.status(200).send({ endpointData });
 };
 
 exports.getArticleById = (req, res, next) => {
   const { article_id } = req.params;
   selectArticleById(article_id)
     .then((article) => {
       if (!article.length) {
-        res.status(404).send({ msg: 'Article does not exist' });
+        res.status(404).send({ msg: "Article does not exist" });
       }
       res.status(200).send({ article });
     })
     .catch((err) => {
-      if(err.code === "22P02") {
-        res.status(400).send({ msg: 'Bad request' });
-      }
-    })
+      next(err);
+    });
 };
 
 exports.getArticles = (req, res, next) => {
-  selectArticles()
-    .then((articles) => {
-      res.status(200).send({ articles });
+  selectArticles().then((articles) => {
+    res.status(200).send({ articles });
+  });
+};
+
+exports.getCommentsById = async (req, res, next) => {
+  const { article_id } = req.params;
+
+  checkArticleExists(article_id)
```

Comment:
Lovely function! However, article_id and author are foreign keys, so PSQL will throw an error (that you can handle with a [PSQL error handler](https://notes.northcoders.com/courses/js-back-end/error-handling#psql-thrown-errors)) if you try and insert a non-existent or invalid value.

This means you don't need to check separately (comment these checks out and run the code to see what I mean - you have one PSQL error code covered in your error handler)